src/cli/format.js

```diff
@@ -460,6 +464,12 @@ async function formatFiles(context) {
       writeOutput(context, result, options);
     }
 
+    if (shouldSetCache) {
+      formatResultsCache?.setFormatResultsCache(filename, options);
+    } else {
+      formatResultsCache?.removeFormatResultsCache(filename);
```

Comment:
Can we just do nothing if the file not written? Will it cause problem?

Reply:
@tksst says, cause problem.
https://github.com/prettier/prettier/pull/13016#issuecomment-1214311862

I'll revert changes.